src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -121,19 +122,82 @@ export default class ComparisonStore {
 
     // < To be implemented in subclasses: >
     public isGroupSelected(name: string): boolean {
-        throw 'isGroupSelected must be implemented in subclass';
+        throw new Error('isGroupSelected must be implemented in subclass');
     }
     public setUsePatientLevelEnrichments(s: boolean) {
-        throw 'setUsePatientLevelEnrichments must be implemented in subclass';
+        throw new Error(
+            'setUsePatientLevelEnrichments must be implemented in subclass'
+        );
+    }
+    public toggleGroupSelected(groupName: string) {
+        throw new Error(`toggleGroupSelected must be implemented in subclass`);
+    }
+    public updateGroupOrder(oldIndex: number, newIndex: number) {
+        throw new Error(`updateGroupOrder must be implemented in subclass`);
+    }
+    public selectAllGroups() {
+        throw new Error(`selectAllGroups must be implemented in subclass`);
+    }
+    public deselectAllGroups() {
+        throw new Error(`deselectAllGroups must be implemented in subclass`);
+    }
+    protected async saveAndGoToSession(newSession: Session) {
+        throw new Error(`saveAndGoToSession must be implemented in subclass`);
+    }
+
+    public get isLoggedIn() {
+        return this.appStore.isLoggedIn;
+    }
+
+    public async addGroup(group: SessionGroupData, saveToUser: boolean) {
+        this.newSessionPending = true;
+        if (saveToUser && this.isLoggedIn) {
+            await comparisonClient.addGroup(group);
+        }
+        const newSession = _.cloneDeep(this._session.result!);
+        newSession.groups.push(group);
+
+        this.saveAndGoToSession(newSession);
     }
 
+    public async deleteGroup(name: string) {
+        this.newSessionPending = true;
+        const newSession = _.cloneDeep(this._session.result!);
+        newSession.groups = newSession.groups.filter(g => g.name !== name);
+
+        this.saveAndGoToSession(newSession);
+    }
+
+    readonly _session: MobxPromise<Session>; // must be implemented in subclasses
     readonly _originalGroups: MobxPromise<ComparisonGroup[]>; // must be implemented in subclasses
     public overlapStrategy: OverlapStrategy; // must be implemented in subclasses
     public usePatientLevelEnrichments: boolean; // must be implemented in subclasses
     readonly samples: MobxPromise<Sample[]>; // must be implemented in subclass
     readonly studies: MobxPromise<CancerStudy[]>; // must be implemented in subclass
     // < / >
 
+    readonly origin = remoteData({
```

Comment:
what is origin again?  maybe a comment or more desriptive name

Reply:
I'll add a comment. it's an existing idiom from the session service, it means the studies that the session (or in this case, the comparison groups) come from